web/src/compose_validate.ts

```diff
@@ -795,3 +796,92 @@ export function convert_mentions_to_silent_in_direct_messages(
     const silent_mention_text = people.get_mention_syntax(full_name, user_id, true);
     return silent_mention_text;
 }
+
+export const SPLIT_DELIMITER = "\n\n\n";
+
+function delimiter_index_outside_code_block(message_content: string): number {
+    let index = message_content.indexOf(SPLIT_DELIMITER);
+    while (index !== -1) {
+        if (
+            !compose_ui.position_inside_code_block(message_content, index + 1) &&
+            !compose_ui.position_inside_code_block(message_content, index + 2)
+        ) {
+            return index;
+        }
+        index = message_content.indexOf(SPLIT_DELIMITER, index + 1);
+    }
+    return -1;
+}
+/**
+ * Splits a message into 2 parts, if possible
+ * @param message_content the content to be split
+ * @param split_messages whether the option is enabled
+ * @returns a pair of strings: the first being the individual split message, the second being the
+ * remaining part of the message, which will be undefined when there are no further parts to send
+ */
+export function split_message(
+    message_content: string,
+    split_messages: boolean,
```

Comment:
Here too, especially with the near identical function name, the parameter should be clearly named

Reply:
Done